File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/duckdb/DateTimeTypes.kt
package dev.typr.foundationskt.docs.duckdb

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*
import java.time.Duration
import java.time.Instant
import java.time.LocalDate
import java.time.LocalDateTime

@Suppress("unused")
class DateTimeTypes {
    //start
    val dateType: DuckDbType<LocalDate> = DuckDbTypes.date
    val tsType: DuckDbType<LocalDateTime> = DuckDbTypes.timestamp
    val tstzType: DuckDbType<Instant> = DuckDbTypes.timestamptz
    val intervalType: DuckDbType<Duration> = DuckDbTypes.interval
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/duckdb/DateTimeTypes.scala
package dev.typr.foundationssc.docs.duckdb
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import java.time.{Duration, Instant, LocalDate, LocalDateTime}

@SuppressWarnings(Array("unused"))
object DateTimeTypes:
  // start
  val dateType: DuckDbType[LocalDate] = DuckDbTypes.date
  val tsType: DuckDbType[LocalDateTime] = DuckDbTypes.timestamp
  val tstzType: DuckDbType[Instant] = DuckDbTypes.timestamptz
  val intervalType: DuckDbType[Duration] = DuckDbTypes.interval
  // stop
